tools/validation-generator/src/main/java/io/spine/tools/validate/code/ViolationTemplate.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.validate.code;
+
+import com.squareup.javapoet.CodeBlock;
+import io.spine.base.FieldPath;
+import io.spine.code.proto.FieldDeclaration;
+import io.spine.protobuf.TypeConverter;
+import io.spine.type.MessageType;
+import io.spine.validate.ConstraintViolation;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.Arrays.asList;
+
+/**
+ * An {@link Expression} which creates a new {@link ConstraintViolation}.
+ */
+public final class ViolationTemplate implements Expression<ConstraintViolation> {
+
+    private final String message;
+    private final @Nullable Expression fieldValue;
+    private final MessageType type;
+    private final FieldPath field;
+    private final @Nullable Expression<Iterable<ConstraintViolation>> nestedViolations;
+
+    private ViolationTemplate(Builder builder) {
+        this.message = builder.message;
+        this.fieldValue = builder.fieldValue;
+        this.type = builder.type;
+        this.field = builder.field;
+        this.nestedViolations = builder.nestedViolations;
+    }
+
+    @Override
+    public CodeBlock toCode() {
+        CodeBlock.Builder builder = CodeBlock
+                .builder()
+                .add("$T.newBuilder()", ConstraintViolation.class)
```

Comment:
Here we use `ConstraintViolation.class` which looks as the duplication of the fact that this class is an expression of type `ConstraintViolation`. Perhaps, we can have something like `classOf()` in `Expression` so that the `<R>` parameter becomes more useful?

Reply:
I intentionally reserved from adding a `GenericTypeIndex` for an `Expression`. I'd like to keep it as simple as possible and avoid exposing extra reflective API where it's not necessary.